### DISCUSSION TITLE
2FA and FIDO support

### DISCUSSION THREAD

**Discussion by caniko** [fastapi-users/fastapi-users#1066] [Ideas]
Hello

In the age of passwords getting stolen or "hacked", I think it is OK to implement 2FA and FIDO to this library. I am hoping it is not too much work:

- `Optional[str]` field on the `User` model for the private key
- The login  included the 2FA or FIDO key.
- Authentication based on `None` (existing way), 2FA, or FIDO

I'd love to hear your opinion on this matter.

Thank you
Can

**Comment by gpkc** [comment]
I have added a comment in another thread with a suggestion on how to implement this: https://github.com/fastapi-users/fastapi-users/discussions/1389#discussioncomment-11520301
